pkg/controller/nodes/dynamic/handler.go

```diff
@@ -448,7 +450,7 @@ func (d dynamicNodeTaskNodeHandler) progressDynamicWorkflow(ctx context.Context,
 		}
 
 		return handler.DoTransition(handler.TransitionTypeEphemeral, handler.PhaseInfoRunning(nil)),
-			handler.DynamicNodeState{Phase: v1alpha1.DynamicNodePhaseFailing, Reason: state.Err.Error()},
+			handler.DynamicNodeState{Phase: v1alpha1.DynamicNodePhaseFailing, Reason: utils.ToString(state.Err)},
```

Comment:
I think you will have to preserve the Error here too (not serialize it) because that's what should be recorded as the error on the dynamic node?

Reply:
removed